pkg/controllers/management/clusterdeploy/clusterdeploy_test.go

```diff
@@ -0,0 +1,104 @@
+package clusterdeploy
+
+import (
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestClusterDeploy_cluster_deploy(t *testing.T) {
+	testClusterAgentToleration := []corev1.Toleration{{
+		Effect: "NoSchedule",
+		Key:    "node-role.kubernetes.io/controlplane-test",
+		Value:  "true",
+	},
+	}
+	testUpdatedClusterAgentToleration := []corev1.Toleration{{
+		Effect: "NoSchedule",
+		Key:    "node-role.kubernetes.io/controlplane-test",
+		Value:  "true",
+	}, {
+		Effect: "NoSchedule",
+		Key:    "node-role.kubernetes.io/etcd-test",
+		Value:  "true",
+	},
+	}
+
+	tests := []struct {
+		name    string
+		cluster *v3.Cluster
+	}{
+		{
+			name: "test-default",
+			cluster: &v3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-default",
+				},
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{},
+				},
+				Status: v3.ClusterStatus{
+					AppliedAgentEnvVars: settings.DefaultAgentSettingsAsEnvVars(),
+				},
+			},
+		},
+		{
+			name: "test-add-cluster-agent-customization",
+			cluster: &v3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-add-cluster-agent-customization",
+				},
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{},
+				},
+				Status: v3.ClusterStatus{
+					AppliedClusterAgentDeploymentCustomization: &v3.AgentDeploymentCustomization{
+						AppendTolerations: testClusterAgentToleration,
+					},
+				},
+			},
+		},
+		{
+			name: "test-update-cluster-agent-customization",
+			cluster: &v3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-default",
+				},
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{
+						ClusterAgentDeploymentCustomization: &v3.AgentDeploymentCustomization{
+							AppendTolerations: testClusterAgentToleration,
+						},
+					},
+				},
+				Status: v3.ClusterStatus{
+					AppliedClusterAgentDeploymentCustomization: &v3.AgentDeploymentCustomization{
+						AppendTolerations: testUpdatedClusterAgentToleration,
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			v3.ClusterConditionAgentDeployed.Message(tt.cluster, "Successfully rolled out agent")
+			tt.cluster.Status.Conditions[0].Status = "True"
+
+			doRedeploy := redeployAgent(tt.cluster, "", "", nil, nil)
+
+			switch tt.name {
+			case "test-default":
```

Comment:
This switch statement can be avoided if we add an additional field to the tests slice. A new bool field could be added, named along the lines of `wantRedeploy`,  then you can simply check against that value for each test case 
```
assert.Equal(t, tt.wantRedeploy, doRedeploy)
```